### PR TITLE
Fixdeprecations

### PR DESCRIPTION
**Pull request by carstenartur**
## What it does
Fix some deprecations by doing what the deprecation message suggests.

## How to test
Ideally nothing should change in behavior.

## Author checklist

- [ ] I have thoroughly tested my changes
- [X] The change is following the [coding conventions](https://wiki.eclipse.org/Platform/How_to_Contribute#Coding_Conventions)
- [X] I have signed the [Eclipse Contributor Agreement (ECA)](https://www.eclipse.org/legal/ECA.php)

### CODE REVIEW

**Comment by carstenartur** [comment]
Hi @jjohnstn , when you find the time could you have a look? While we reduced the amount of warnings with the last patch by half we are still one of the projects with the highest amount of deprecations. If you are unsure about one or another change feel free to leave it out this time.
image

**Comment by jjohnstn** [comment]
Hi @carstenartur Will look at it today.

**Review by jjohnstn — CHANGES_REQUESTED** [review]

File: org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/JavaManipulationPlugin.java [hunk 1]
```diff
@@ -173,12 +174,18 @@ public Bundle[] getBundles(String bundleName, String version) {
 		if (bundles != null)
 			return bundles;
 
-		// Accessing unresolved bundle
-		ServiceReference<PackageAdmin> serviceRef= fBundleContext.getServiceReference(PackageAdmin.class);
-		PackageAdmin admin= fBundleContext.getService(serviceRef);
-		bundles= admin.getBundles(bundleName, version);
-		if (bundles != null && bundles.length > 0)
-			return bundles;
+		bundles = fBundleContext.getBundles();
```
[the remark this answers is not in the document]
Comment (carstenartur) [reply org.eclipse.jdt.core.manipulation/common/org/eclipse/jdt/internal/core/manipulation/JavaManipulationPlugin.java]: As I failed to find official Update documentation I removed it.

**Comment by akurtakov** [comment]
Would you guys please the failures at check https://download.eclipse.org/eclipse/downloads/drops4/I20241020-0600/testresults/html/org.eclipse.pde.ui.tests_ep434I-unit-cen64-gtk3-java23_linux.gtk.x86_64_23.html ? They started to appear after this change and I haven't noticed any other change in this area.

**Comment by carstenartur** [comment]
Just from looking at the code 
right now I cannot see where the issue is. Can someone rollback and maybe next Weekend I can have a look?

**Comment by akurtakov** [comment]
Looks like it has different cause as explained in https://github.com/eclipse-jdt/eclipse.jdt.core/issues/3125#issuecomment-2425807445

**Pull request merged**